src/common/tests_dmg_helpers.c

```diff
@@ -487,6 +488,19 @@ dmg_pool_create(const char *dmg_config_file,
 					    entry->dpe_str);
 			if (args == NULL)
 				D_GOTO(out, rc = -DER_NOMEM);
+		} else {
+			static bool need_init_rand = true;
+
+			if (need_init_rand) {
```

Comment:
can you explain when this would be false? I don't understand this block

Reply:
Ah, yeah, this is a C-specific thing. Since it's a static variable, the value will remain in memory even when it goes out of scope. It was initialized as true, but after it's set to false once, that value will stick for the next caller.

Reply:
so why not initialize need_init_rand at the top of the function, isn't that part of the coding style guidelines?

Reply:
The current guidance is to declare the variable at the top of the scope where it's used, especially if it has a very limited scope like this one. But even if it was initialized at the top, it would still need to be static to ensure we didn't re-seed on every pool create call.